mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaTextExtractorTest.java

```diff
@@ -121,5 +151,76 @@ public void excelTest() throws Exception {
                 "\n" +
                 "\n");
     }
-    
+
+    @Test
+    public void deserializerShouldThrowWhenMoreThanOneNode() throws Exception {
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("The response should have only one element");
+
+        TikaTextExtractor textExtractor = new TikaTextExtractor(new TikaHttpClient() {
+            
+            @Override
+            public InputStream rmetaAsJson(InputStream inputStream, String contentType) throws TikaException {
+                return new ByteArrayInputStream("[{\"key1\":\"value1\"},{\"key2\":\"value2\"}]".getBytes(Charsets.UTF_8));
+            }
+        });
+
+        InputStream inputStream = null;
+        textExtractor.extractContent(inputStream, "text/plain", "fake.txt");
+    }
+
+    @Test
+    public void deserializerShouldThrowWhenNodeIsNotAnObject() throws Exception {
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("The element should be a Json object");
+
+        TikaTextExtractor textExtractor = new TikaTextExtractor(new TikaHttpClient() {
+            
+            @Override
+            public InputStream rmetaAsJson(InputStream inputStream, String contentType) throws TikaException {
+                return new ByteArrayInputStream("[\"value1\"]".getBytes(Charsets.UTF_8));
+            }
+        });
+
+        InputStream inputStream = null;
+        textExtractor.extractContent(inputStream, "text/plain", "fake.txt");
+    }
+
+    @Test
+    public void asListOfStringShouldReturnASingletonWhenOneElement() {
```

Comment:
"Singleton" means "single element"

Exemple from java API:

```
        Collections.singletonList(new Object())
```

Si IMO naming here is OK.